eclair-core/src/main/scala/fr/acinq/eclair/json/JsonSerializers.scala

```diff
@@ -221,13 +221,19 @@ class ColorSerializer extends CustomSerializerOnly[Color](_ => {
   case c: Color => JString(c.toString)
 })
 
-class RouteResponseSerializer extends CustomSerializerOnly[RouteResponse](_ => {
-  case route: RouteResponse =>
-    val nodeIds = route.routes.head.hops match {
-      case rest :+ last => rest.map(_.nodeId) :+ last.nodeId :+ last.nextNodeId
-      case Nil => Nil
-    }
-    JArray(nodeIds.toList.map(n => JString(n.toString)))
+class RouteSerializer extends CustomSerializerOnly[Route](_ => {
+  case route: Route =>
+    JObject(
+      ("amount", JLong(route.amount.toLong)),
+      ("hops", JArray(route.hops.map(hop => Extraction.decompose(hop)(
```

Comment:
Does that mean you're printing the whole `channel_update` for each hop?
Is that really necessary?
Most of the fields don't make sense (e.g. signature, chainhash), it would be better to have a more fine-grained choice of what exactly we print.

Reply:
1. that's the only way to get a channel ID. 
2. channel updates provide fee rates which are used for figuring out economical viability of found routes
3. one can get channel updates using `allupdates`, but without a channel ID it's unclear how to get an update when two nodes have more than one channel between each other. 
4. since there is an update for every hop, there's no need to call `allupdates` at all

Reply:
Let's wait for https://github.com/ACINQ/eclair/pull/1979 to be integrated, this will give you an opportunity to more easily fine-tune the serialization you want to use for this case.